src/app/infrastructure/RandData.js

```diff
@@ -0,0 +1,47 @@
+import {PrepareData} from './processData'
+import {Data} from './fetchApi'
+import _ from 'lodash'
+
+const num = 10
+
+class RandData {
+     async randData(category) {
+        const data = new Data();
+        const prepareData = new PrepareData();
+        let categoryArr = []
+        switch (category) {
+            case 'people':
+                categoryArr = await data.getPeopleJsonData().then(res => prepareData.preprocessData(res) )
+                break;
+            case 'starships':
+                categoryArr = await data.getStarshipsJsonData().then(res => prepareData.preprocessData(res) )
+                break;
+            case 'vehicles':
+                categoryArr = await data.getVehiclesJsonData().then(res => prepareData.preprocessData(res) )
+            default:
+                categoryArr = await data.getPeopleJsonData().then(res => prepareData.preprocessData(res) )
```

Comment:
W przypadku podania złej kategorii powinniśmy rzucać wyjątek, ponieważ w tym przypadku oznacza to, że mamy błąd w kodzie i łatwiej będzie go znaleźć kiedy rzucimy `Error`, niż gdy wykona się z losową wartością (wtedy dużo trudniej jest zidentyfikować gdzie jest problem) .